server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/Dequeuer.java

```diff
@@ -141,17 +140,25 @@ public void close() {
         };
     }
 
-    private MailWithEnqueueId loadMail(Delivery response) throws MailQueue.MailQueueException {
-        MailReferenceDTO mailDTO = toMailReference(response);
-        return mailLoader.apply(mailDTO);
+    private Mono<MailWithEnqueueId> loadMail(AcknowledgableDelivery response) {
+        return toMailReference(response)
+            .flatMap(reference -> mailLoader.load(reference)
+                .onErrorResume(ObjectNotFoundException.class, e -> {
+                    LOGGER.error("Fail to load mail {} with enqueueId {} as underlying blobs do not exist. Discarding this message to prevent an infinite loop.", reference.getName(), reference.getEnqueueId(), e);
+                    response.nack(!REQUEUE);
+                    return Mono.empty();
+                })
+                .onErrorResume(e -> {
+                    LOGGER.error("Fail to load mail {} with enqueueId {}", reference.getName(), reference.getEnqueueId(), e);
```

Comment:
I think we still don't want to requeue it if the JSON is invalid.

Reply:
I tend to agree with this statement. However there's disasters waiting to happen on dto format changes. If we consider it won't happen because we wont do the mistake and anyway will provide a mechanism to rebuild rabbit content.